### PR TITLE
dump envoy config_dump and clusters info to perf GCS bucket

### PR DESCRIPTION
**Pull request by carolynhu**
Gather and dump envoy config_dump and clusters info to perf GCS bucket.

### CODE REVIEW

File: perf/benchmark/run_benchmark_job.sh [hunk 1]
```diff
@@ -111,6 +111,46 @@ function setup_fortio_and_prometheus() {
...
+
+function kill_envoy_port() {
+    # shellcheck disable=SC2155
+    kill $(lsof -t -i:15000)
+}
+
+function collect_config_dump() {
+  kill_envoy_port
+  port_forward_envoy "${FORTIO_CLIENT_POD}"
+  FORTIOCLIENT_CONFIG_DUMP="${FORTIO_CLIENT_POD}"_config_dump.yaml
```
Comment (mandarjog) [inline perf/benchmark/run_benchmark_job.sh:125]: This is a lot of repeated code in 2 fns, refactor?

```bash
function collect (url, dump_dir, file_name_suffix) {
}
```
and then
``` bash
collect "/config_dump" "config_dumps" _config_dump.yaml 
```

File: perf/benchmark/run_benchmark_job.sh [hunk 2]
```diff
@@ -111,6 +111,45 @@ function setup_fortio_and_prometheus() {
...
+  kubectl -n "${NAMESPACE}" port-forward "${1}" 15000 &
+}
+
+function kill_envoy_port() {
+    kill "$(lsof -t -i:15000)"
+}
+
+function collect_config_dump() {
+  kill_envoy_port
+  port_forward_envoy "${FORTIO_CLIENT_POD}"
```
Comment (howardjohn) [inline perf/benchmark/run_benchmark_job.sh:124]: might be easier to just `kubectl exec`?

File: perf/benchmark/run_benchmark_job.sh (hunk 1 above)
Comment (carolynhu) [reply perf/benchmark/run_benchmark_job.sh:125]: yeah, sure!

File: perf/benchmark/run_benchmark_job.sh [hunk 3]
```diff
@@ -111,6 +111,36 @@ function setup_fortio_and_prometheus() {
...
+    kill "$(lsof -t -i:15000)"
+}
+
+function collect_envoy_info() {
+  POD_NAME=${1}
+  FILE_SUFFIX=${2}
+
+  kill_envoy_port
+  port_forward_envoy "${POD_NAME}"
+  ENVOY_DUMP_NAME="${POD_NAME}"_"${FILE_SUFFIX}".yaml
```
Comment (mandarjog) [inline perf/benchmark/run_benchmark_job.sh:128]: this should be 
`"${POD_NAME}_${FILE_SUFFIX}.yaml"`

File: perf/benchmark/run_benchmark_job.sh (hunk 2 above)
Comment (mandarjog) [reply perf/benchmark/run_benchmark_job.sh]: @carolynhu I agree kubectl exec does not involve any races or waiting like port forward.

File: perf/benchmark/run_benchmark_job.sh (hunk 3 above)
Comment (carolynhu) [reply perf/benchmark/run_benchmark_job.sh]: done!

File: perf/benchmark/run_benchmark_job.sh (hunk 2 above)
Comment (carolynhu) [reply perf/benchmark/run_benchmark_job.sh]: updated to use kubectl exec

**Pull request merged**